pdp/context.go

```diff
@@ -192,6 +202,41 @@ func (c *Context) calculateStringExpression(e Expression) (string, error) {
 	return v.str()
 }
 
+func (c *Context) calculateIntegerExpression(e Expression) (int64, error) {
+	v, err := e.Calculate(c)
+	if err != nil {
+		return 0, err
+	}
+
+	return v.integer()
+}
+
+func (c *Context) calculateFloatExpression(e Expression) (float64, error) {
+	v, err := e.Calculate(c)
+	if err != nil {
+		return 0, err
+	}
+
+	return v.float()
+}
+
+func (c *Context) calculateFloatOrIntegerExpression(e Expression) (float64, error) {
```

Comment:
I'd like to see explicit conversion for each type. This assumes int-to-float implicit conversion. Having expression by argument type function selection on parsing stage such implicit conversion makes hard to follow function resolution rules. And we need to document it clearly as C++ does. I'd like to not mess with it.

Reply:
I understand your concern.

I just wanted to point out that `calculateFloatOrIntegerExpression` is specifically designed for the float numerical functions, where I specifically want to promote int arguments to float. Without it, the code will be duplicated for each of the arguments for each of the float functions.

Maybe I can move/rename that function if you feel it doesn't below to that group of `calculate` functions.

Please let me know how you'd like to proceed.